asr/wenet/cli/reverb.py

```diff
@@ -0,0 +1,392 @@
+import copy
+from functools import cached_property, lru_cache, partial
+import logging
+import os
+from itertools import groupby, chain
+from pathlib import Path
+from math import ceil
+from typing import Generator, List
+import shutil
+import yaml
+
+import torch
+import torch.nn.functional as F
+
+import torchaudio
+from torchaudio.compliance import kaldi
+
+from wenet.cli.utils import hyps_to_ctm, hyps_to_txt
+from wenet.transformer.cmvn import GlobalCMVN
+from wenet.transformer.search import DecodeResult
+from wenet.text.rev_bpe_tokenizer import RevBpeTokenizer
+from wenet.utils.cmvn import load_cmvn
+from wenet.utils.init_model import init_model
+from wenet.utils.init_tokenizer import init_tokenizer
+from wenet.utils.ctc_utils import get_blank_id
+from wenet.bin.ctc_align import ctc_align, adjust_model_time_offset
+
+
+CACHED_MODELS_DIR = Path.home() / ".cache/reverb"
+_MODELS = {
+    "reverb_asr_v1": "https://huggingface.co/Revai/reverb-asr",
+}
+
+
+class ReverbASR:
+    def __init__(self,
+        config,
+        checkpoint,
+        *,
```

Comment:
Originally I thought it would be safer to force the user to specify which file they wish to override -- after discussing and rethinking, I'll remove this and allow for positional or keyword passing the subsequent arguments